mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -0,0 +1,161 @@
+      .of(InputFormatConfig.TABLE_SCHEMA, InputFormatConfig.PARTITION_SPEC, Catalogs.LOCATION, Catalogs.NAME)
+      .collect(Collectors.toCollection(HashSet::new));
+  private static final Set<String> PROPERTIES_TO_REMOVE = Stream
+      .of(InputFormatConfig.HIVE_DELETE_BACKING_TABLE, "storage_handler", "EXTERNAL")
+      .collect(Collectors.toCollection(HashSet::new));
+
+  private final Configuration conf;
+  private Table icebergTable = null;
+  private Properties catalogProperties;
+  private boolean deleteIcebergTable;
+
+  public HiveIcebergMetaHook(Configuration conf) {
+    this.conf = conf;
+  }
+
+  @Override
+  public void preCreateTable(org.apache.hadoop.hive.metastore.api.Table hmsTable) {
+    catalogProperties = getCatalogProperties(hmsTable);
+    try {
+      icebergTable = Catalogs.loadTable(conf, catalogProperties);
+
+      Preconditions.checkArgument(catalogProperties.getProperty(InputFormatConfig.TABLE_SCHEMA) == null,
+          "Iceberg table already created - can not use provided schema");
+      Preconditions.checkArgument(catalogProperties.getProperty(InputFormatConfig.PARTITION_SPEC) == null,
+          "Iceberg table already created - can not use provided partition specification");
+
+      LOG.info("Iceberg table already exists {}", icebergTable);
+    } catch (NoSuchTableException nte) {
+      String schemaString = catalogProperties.getProperty(InputFormatConfig.TABLE_SCHEMA);
+      Preconditions.checkNotNull(schemaString, "Please provide a table schema");
+      // Just check if it is parsable, and later use for partition specification parsing
+      Schema schema = SchemaParser.fromJson(schemaString);
+
+      String specString = catalogProperties.getProperty(InputFormatConfig.PARTITION_SPEC);
+      if (specString != null) {
+        // Just check if it is parsable
+        PartitionSpecParser.fromJson(schema, schemaString);
+      }
+
+      // Allow purging table data if the table is created now and not set otherwise
+      if (hmsTable.getParameters().get(InputFormatConfig.HIVE_DELETE_BACKING_TABLE) == null) {
+        hmsTable.getParameters().put(InputFormatConfig.HIVE_DELETE_BACKING_TABLE, "TRUE");
+      }
+
+      // Set the table type even for non HiveCatalog based tables
+      hmsTable.getParameters().put(BaseMetastoreTableOperations.TABLE_TYPE_PROP,
+          BaseMetastoreTableOperations.ICEBERG_TABLE_TYPE_VALUE.toUpperCase());
+
+      // Remove creation related properties
+      PARAMETERS_TO_REMOVE.forEach(hmsTable.getParameters()::remove);
+    }
+  }
+
+  @Override
+  public void rollbackCreateTable(org.apache.hadoop.hive.metastore.api.Table hmsTable) {
+    // do nothing
+  }
+
+  @Override
+  public void commitCreateTable(org.apache.hadoop.hive.metastore.api.Table hmsTable) {
+    if (icebergTable == null) {
+      catalogProperties.put(HiveTableOperations.TABLE_FROM_HIVE, true);
+      LOG.info("Iceberg table creation with the following properties {}", catalogProperties.keySet());
```

Comment:
Isn't this already logged in the catalog implementation?

Reply:
Removed